tests/unit/s2n_tls13_handshake_early_data_test.c

```diff
@@ -0,0 +1,608 @@
+        }
+
+        /* Test all client cases */
+        for (size_t state_i = 0; state_i < s2n_array_len(client_early_data_transitions); state_i++) {
+            for (uint16_t handshake_type = 0; handshake_type < S2N_HANDSHAKES_COUNT; handshake_type++) {
+                EXPECT_OK(s2n_test_key_schedule(S2N_CLIENT, handshake_type, client_early_data_transitions[state_i],
+                        &valid_test_case));
+            }
+        }
+
+        /* Test all server_cases */
+        for (size_t state_i = 0; state_i < s2n_array_len(server_early_data_transitions); state_i++) {
+            for (uint16_t handshake_type = 0; handshake_type < S2N_HANDSHAKES_COUNT; handshake_type++) {
+                EXPECT_OK(s2n_test_key_schedule(S2N_SERVER, handshake_type, server_early_data_transitions[state_i],
+                        &valid_test_case));
+            }
+        }
+    }
+
+    /* Test early data encryption */
+    {
+        /**
+         *= https://tools.ietf.org/rfc/rfc8448#section-3
+         *= type=test
+         *# {server}  generate resumption secret "tls13 resumption":
+         *#
+         *#    PRK (32 octets):  7d f2 35 f2 03 1d 2a 05 12 87 d0 2b 02 41 b0 bf
+         *#       da f8 6c c8 56 23 1f 2d 5a ba 46 c4 34 ec 19 6c
+         *#
+         *#    hash (2 octets):  00 00
+         *#
+         *#    info (22 octets):  00 20 10 74 6c 73 31 33 20 72 65 73 75 6d 70 74
+         *#       69 6f 6e 02 00 00
+         *#
+         *#    expanded (32 octets):  4e cd 0e b6 ec 3b 4d 87 f5 d6 02 8f 92 2c
+         *#       a4 c5 85 1a 27 7f d4 13 11 c9 e6 2d 2c 94 92 e1 c4 f3
+         */
+        S2N_BLOB_FROM_HEX(psk_secret,"4e cd 0e b6 ec 3b 4d 87 f5 d6 02 8f 92 2c \
+                  a4 c5 85 1a 27 7f d4 13 11 c9 e6 2d 2c 94 92 e1 c4 f3");
+
+        /**
+         *= https://tools.ietf.org/rfc/rfc8448#section-3
+         *= type=test
+         *# {server}  construct a NewSessionTicket handshake message:
+         *#
+         *#    NewSessionTicket (205 octets):  04 00 00 c9 00 00 00 1e fa d6 aa
+         *#       c5 02 00 00 00 b2 2c 03 5d 82 93 59 ee 5f f7 af 4e c9 00 00 00
+         *#       00 26 2a 64 94 dc 48 6d 2c 8a 34 cb 33 fa 90 bf 1b 00 70 ad 3c
+         *#       49 88 83 c9 36 7c 09 a2 be 78 5a bc 55 cd 22 60 97 a3 a9 82 11
+         *#       72 83 f8 2a 03 a1 43 ef d3 ff 5d d3 6d 64 e8 61 be 7f d6 1d 28
+         *#       27 db 27 9c ce 14 50 77 d4 54 a3 66 4d 4e 6d a4 d2 9e e0 37 25
+         *#       a6 a4 da fc d0 fc 67 d2 ae a7 05 29 51 3e 3d a2 67 7f a5 90 6c
+         *#       5b 3f 7d 8f 92 f2 28 bd a4 0d da 72 14 70 f9 fb f2 97 b5 ae a6
+         *#       17 64 6f ac 5c 03 27 2e 97 07 27 c6 21 a7 91 41 ef 5f 7d e6 50
+         *#       5e 5b fb c3 88 e9 33 43 69 40 93 93 4a e4 d3 57 00 08 00 2a 00
+         *#       04 00 00 04 00
+         */
+        S2N_BLOB_FROM_HEX(psk_identity,
+                                   "2c 03 5d 82 93 59 ee 5f f7 af 4e c9 00 00 00 \
```

Comment:
At first glance the value looked wrong but I'm assuming the trimmed bytes include framing overhead? Can you add a comment explaining why those bytes are trimmed?